LinkedList/Exercise/SLList.h

```diff
@@ -0,0 +1,328 @@
+		for (int i = 0; i < InList.ListSize; i++)
+		{
+			AddTail(Current->Element);
+			Current = Current->Next;
+		}
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	Type& operator[] (const int Index)
+	{
+		return GetNode(Index)->Element;
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	void operator= (TSLList& InList)
+	{
+		Clear();
+		FNode* Current = InList.Head;
+		for (int i = 0; i < InList.ListSize; i++)
+		{
+			AddTail(Current->Element);
+			Current = Current->Next;
+		}
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	const Type& operator= (const int Index) const
+	{
+		return GetNode(Index)->Element;
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	Type& GetHead()
+	{
+		return Head->Element;
+	}
+
+	const Type& GetHead() const
+	{
+		return Head->Element;
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	Type& GetTail()
+	{
+		return Tail->Element;
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	const Type& GetTail() const
+	{
+		return Tail->Element;
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	bool IsEmpty() const
+	{
+		return ListSize == 0;
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	int GetSize() const
+	{
+		return ListSize;
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	void AddHead(const Type& InElement)
+	{
+		FNode* NewHead = new FNode();
+		NewHead->Element = InElement;
+		NewHead->Next = Head;
+
+		Head = NewHead;
+
+		if (ListSize == 0)
+		{
+			Tail = Head;
+		}
+
+		ListSize++;
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	void AddTail(const Type& InElement)
+	{
+		if (ListSize == 0)
+		{
+			AddHead(InElement);
+		}
+		else
+		{
+			FNode* NewTail = new FNode();
+			NewTail->Element = InElement;
+
+			Tail->Next = NewTail;
+			Tail = NewTail;
+
+			ListSize++;
+		}
+
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	void Insert(const Type& InElement, const int Index)
+	{
+		if (Index == 0)
+		{
+			AddHead(InElement);
+		}
+		else if (Index > 0 && Index < ListSize)
+		{
+			FNode* Current = GetNode(Index);
+
+			FNode* NewNode = new FNode();
+			NewNode->Element = InElement;
+
+			GetNode(Index - 1)->Next = NewNode;
+			NewNode->Next = Current;
+			ListSize++;
+		}
+	}
+
+	//-------------------------------------------------------------------------------------
+
+	void Remove(const int Index)
+	{
+		if (Index == 0)
+		{
+			FNode* NextUp = Head->Next;
+			delete Head;
+			Head = NextUp;
+		}
+		else if (Index > 0 && Index < ListSize)
```

Comment:
Same as before, returning in case of index out of bounds at the beginning of the function is a good option